batch_object_tagger.tf

```
@@ -0,0 +1,147 @@
+locals {
+  pdm_object_tagger_image = "${local.account.management}.${data.terraform_remote_state.aws_ingestion.outputs.vpc.vpc.ecr_dkr_domain_name}/dataworks-s3-object-tagger:${var.image_version.s3-object-tagger}"
+  pdm_object_tagger_application_name = "pdm-s3-object-tagger"
+}
+
+# AWS Batch Job IAM role
+data "aws_iam_policy_document" "batch_assume_policy" {
+  statement {
+    sid    = "BatchAssumeRolePolicy"
+    effect = "Allow"
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      identifiers = ["ecs-tasks.amazonaws.com"]
+
+      type = "Service"
+    }
+  }
+}
+
+resource "aws_iam_role" "pdm_object_tagger" {
+  name               = "pdm_object_tagger"
+  assume_role_policy = data.aws_iam_policy_document.batch_assume_policy.json
+  tags               = local.common_tags
+}
+
+data "aws_iam_policy_document" "pdm_object_tagger_config_bucket" {
+  statement {
+    sid    = "AllowS3Tagging"
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:GetObjectTagging",
+      "s3:PutObjectTagging"
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.common.outputs.config_bucket.arn}/*",
```

can you specify the path? I think it's this:
```suggestion
      "${data.terraform_remote_state.common.outputs.config_bucket.arn}/component/rbac/*",
```

---

I disagree with this. As the path is configurable in the pipeline for the config bucket, this may be changed on the fly.

---

that container shouldn't have full access to that config bucket

---

List has to be on the entire bucket, so I would agree with Dan on splitting the objects stuff.

---

I've made the split but the addition of component/rbac/* to the bucket rather than /* will limit the location of the config CSV going forward. It's currently configurable in the pipeline per env